packages/riverpod_lint/README.md

```diff
@@ -643,6 +644,91 @@ switch (...) {
 }
 ```
 
+### incorrect_usage_of_ref_method
+
+Warn if the `ref.watch`/`ref.read`/`ref.listen`/`ref.listenManual` methods are used incorrectly.
+
+**Good**:
+
+```dart
+final fnProvider = Provider<int>((ref) {
+  ref.watch(provider);
+  ref.listen(provider, ...);
+  return 0;
+});
+
+class MyNotifier extends Notifier<int> {
+  int get _readGetter => ref.read(provider);
+
+  @override
+  int build() {
+    ref.watch(provider);
+    ref.listen(provider, ...);
+    return 0;
+  }
+
+  void someMethod() {
+    ref.read(provider);
+  }
+}
+
+void initState() {
+  ref.listenManual(provider, ...);
+}
+
+Widget build(ctx, ref) {
+  ref.watch(provider);
+  ref.listen(provider, ...);
+
+  return Button(
+    onPressed: () {
+      ref.read(provider);
+      ref.listenManual(provider, ...);
+    }
+  );
+}
+```
+
+**Bad**:
+
+```dart
+final fnProvider = Provider<int>((ref) {
+  ref.read(provider); // use watch instead
+  return 0;
+});
+
+class MyNotifier extends Notifier<int> {
+  int get _watchGetter => ref.watch(provider); // use read instead
+
+  @override
+  int build() {
+    ref.read(provider); // use watch instead
+    return 0;
+  }
+
+  void someMethod() {
+    ref.watch(provider); // use read instead
+    ref.listen(provider, ...); // place listen in build instead
```

Comment:
The recommendation would be to use listenManual

Reply:
but this is inside a Notifier, ref is not WidgetRef.

Reply:
Sorry I missed that. The there's no issue with calling listen in methods.
As long as the return value is used to cancel subscription.

Reply:
I see. So this is a valid use case of Ref.listen.
```dart
void someMethod(){
  sub?.close();
  sub = ref.listen(provider, ...);
}
```